### PR TITLE
Add custom model support

### PR DESCRIPTION
**Pull request by quic-shagun**
Added --local_model_dir argument which allows users to pass custom model weights and config files

### CODE REVIEW

**Review by ochougul — APPROVED** [review]
LGTM

File: QEfficient/utils/_utils.py [hunk 1]
```diff
@@ -119,13 +119,14 @@ def onnx_exists(model_name: str) -> Tuple[bool, str, str]:
...
+def load_hf_tokenizer(model_name: str, cache_dir: Optional[str] = None, hf_token: Optional[str] = None, local_model_dir: Optional[str] = None, padding_side:str = "left", **kwargs) -> Union[PreTrainedTokenizerFast, PreTrainedTokenizer]:
     logger.info(f"Loading Tokenizer for {model_name}")
     if hf_token is not None:
         login(hf_token)
-
+    if local_model_dir and cache_dir:
+        logger.warning(f"Both local model dir ({local_model_dir}) and cache dir ({cache_dir}) given. Using local model dir.")
     # Download tokenizer along with model if it doesn't exist
-    model_hf_path = hf_download(repo_id=model_name, cache_dir=cache_dir, allow_patterns=["*.json", "*.py", "*token*"])
+    model_hf_path = local_model_dir if local_model_dir else hf_download(repo_id=model_name, cache_dir=cache_dir, allow_patterns=["*.json", "*.py", "*token*"])
```
Comment (ochougul) [inline QEfficient/utils/_utils.py:129]: Hey @quic-shagun , can we handle this without introducing this new param?
rename `model_name` parameter to `pretrained_model_name_or_path` and if it's a dir, we directly jump to loading, if it's not a dir, we download?

**Review by ochougul — CHANGES_REQUESTED** [review]

File: QEfficient/utils/_utils.py (hunk 1 above)
Comment (quic-shagun) [reply QEfficient/utils/_utils.py:129]: We are logging info here 
```
logger.info(f"Loading Tokenizer for {model_name}")
```
If we make `model_name` as `pretrained_model_name_or_path` then the above info will not make sense for path

Comment (quic-shagun) [reply QEfficient/utils/_utils.py]: Discussed offline and changed logging message

**Comment by vbaddi** [comment]
LGTM, Please resolve conflicts and rebase to the mainline. Thanks

File: QEfficient/cloud/export.py [hunk 2]
```diff
@@ -52,27 +53,31 @@ def get_onnx_model_path(model_name: str, cache_dir: str, tokenizer: Optional[Uni
...
 if __name__ == "__main__":
     parser = argparse.ArgumentParser(description="Export script.")
     parser.add_argument("--model_name", "--model-name", required=True, help="HF Model card name/id")
+    parser.add_argument("--local-model-dir", "--local_model_dir", required=False, help="Path to custom model weights and config files")
     parser.add_argument(
         "--cache_dir",
         "--cache-dir",
         required=False,
-        default=Constants.CACHE_DIR,
+        default=None,
```
Comment (ochougul) [inline QEfficient/cloud/export.py:80]: not required. remove line 80.

File: QEfficient/utils/_utils.py [hunk 3]
```diff
@@ -140,3 +139,9 @@ def get_qpc_dir_name_infer(num_cores, mos, batch_size, prompt_len, ctx_len, mxfp
     )
 
     return qpc_base_dir_name
+
+def check_and_assign_cache_dir(local_model_dir, cache_dir):
+    if local_model_dir is not None and cache_dir is not None:
+        logger.warning(f"Both local model dir ({local_model_dir}) and cache dir ({cache_dir}) given. Using local model dir.")
+    elif cache_dir is None:
+        cache_dir = Constants.CACHE_DIR
```
[the remark this answers is not in the document]
Comment (ochougul) [reply QEfficient/utils/_utils.py:147]: this shoudl be 
```
if local_model_dir is None and cache_dir is None:
    cache_dir = Constants.CACHE_DIR
```

**Review by ochougul — CHANGES_REQUESTED** [review]
Please rebase

File: QEfficient/cloud/export.py (hunk 2 above)
Comment (quic-shagun) [reply QEfficient/cloud/export.py]: Done

File: QEfficient/utils/_utils.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (quic-shagun) [reply QEfficient/utils/_utils.py:147]: Done

File: .gitignore [hunk 4]
```diff
@@ -84,3 +84,7 @@ dmypy.json
 
 # Cython debug symbols
 cython_debug/
+
+# Local Files
+cache_dir
+qeff_models
```
Comment (quic-shagun) [inline .gitignore:90]: Adding these files as we don't need them to be committed

File: QEfficient/utils/_utils.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ochougul) [reply QEfficient/utils/_utils.py:147]: Still pass by value?

File: README.md [hunk 5]
```diff
@@ -193,7 +193,7 @@ python -m QEfficient.cloud.infer --model_name gpt2 --batch_size 1 --prompt_len 3
...
-| QEfficient.export |     [click here](#2-onnx-export-of-transformed-model)       |   <li>mode_name : $\color{green} {Mandatory}$</li><li>model_kv : Optional </li><li>tokenizer : Optional</li><li>onnx_dir_path : Optional </li><li>hf_token : Optional</li><li>seq_length : Optional [Default-128]</li><li>kv : Optional [Default-$\color{green} {True}$]</li><li>form_factor : Optional [Default-"cloud"]</li>| 
+| QEfficient.export |     [click here](#2-onnx-export-of-transformed-model)       |   <li>mode_name : $\color{green} {Mandatory}$</li><li>model_kv : Optional </li><li>local_model_dir : Optional [Path to custom model weights and config file]</li><li>tokenizer : Optional</li><li>onnx_path : Optional </li><li>hf_token : Optional</li><li>seq_length : Optional [Default-128]</li><li>kv : Optional [Default-$\color{green} {True}$]</li><li>return_path : Optional [Default-False]</li><li>form_factor : Optional [Default-"cloud"]</li><li>***save_fp32_onnx : Optional [Default-False]</li><li>***save_fp16_onnx : Optional [Default-True]</li>|
```
Comment (ochougul) [inline README.md:196]: return path and others are already removed. Please check in main.

**Review by ochougul — CHANGES_REQUESTED** [review]
Please run ruff linter once. Thanks.

**Pull request merged**